crates/amaru-ledger/src/context.rs

```diff
@@ -47,7 +47,7 @@ pub trait PreparationContext<'a>:
 // Errors
 // -------------------------------------------------------------------------------------------------
 
-#[derive(thiserror::Error, Debug)]
+#[derive(thiserror::Error, Debug, Clone)]
```

Comment:
Now that's suspicious e_e. 

I purposefully kept that one out because I don't think we should ever need to clone errors?

Reply:
This is a side-effect of `InvalidBlock` being used in `BlockValidationResult`, itself passed across `gasket` stages that introduces this requirement.
The alternative would be to not pass `InvalidBlock` but a `Clone` friendly equivalent, now `InvalidBlock` is pretty complex and it looks useful to benefit from it down the line.

Reply:
I simplified things a bit, your point should be addressed now.